src/QMCWaveFunctions/Fermion/MultiDiracDeterminant.2.cpp

```diff
@@ -792,7 +792,14 @@ void MultiDiracDeterminant::mw_evaluateDetsAndGradsForPtclMove(
         psiV_temp_list_ptr[iw][i] = psiV_list_devptr[iw][J];
         ratioGradRef_local += psiMinv_temp_list_devptr[iw][i * psiMinv_cols + WorkingIndex] * dpsiV_list_ptr[iw][J];
       }
-      ratioGradRef_list_ptr[iw] = ratioGradRef_local;
+
+      // Workaround for https://github.com/QMCPACK/qmcpack/issues/4767
+      // Compiler-generated assignment fails for the second component.
+      // Workaround is to assign each component individually.
+      //ratioGradRef_list_ptr[iw] = ratioGradRef_local;
+      ratioGradRef_list_ptr[iw][0] = ratioGradRef_local[0];
```

Comment:
This fix would be ok if we had no way to actually fix TinyVector, say it was a type from a closed source library.

This should be checked for in TinyVectors unit tests.  And an special assignment method for omptarget would make sense. It should just default to the normal assignment operator if the compiler figures this out properly.  And this needs to be documented in TinyVector so the next person to assign to one in a target section has a fighting chance.  They're not going to see this comment.  The documentation of the special assignment for TinyVector should end up in the doxygen.

Reply:
The underlying issue also affects complex numbers.  I'm trying to make standalone reproducer, but I'm not having much success.